x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts

```diff
@@ -0,0 +1,972 @@
+      success_count: normalizeValue<number>(fields[EVENT_SUCCESS_COUNT]),
+    },
+    processor: {
+      name: normalizeValue<'transaction'>(fields[PROCESOR_NAME]),
+      event: normalizeValue<'span'>(fields[PROCESSOR_EVENT]),
+    },
+    transaction: {
+      id: normalizeValue<string>(fields[TRANSACTION_ID]),
+    },
+    span: {
+      duration: {
+        us: normalizeValue<number>(fields[SPAN_DURATION]),
+      },
+      subtype: normalizeValue<string>(fields[SPAN_SUBTYPE]),
+      name: normalizeValue<string>(fields[SPAN_NAME]),
+      destination: {
+        service: {
+          resource: normalizeValue<string>(fields[SPAN_DESTINATION_SERVICE_RESOURCE]),
+        },
+      },
+      id: normalizeValue<string>(fields[SPAN_ID]),
+      type: normalizeValue<string>(fields[SPAN_TYPE]),
+      representative_count: normalizeValue<number>(fields[SPAN_REPRESENTATIVE_COUNT]),
+    },
+    timestamp: {
+      us: normalizeValue<number>(fields[TIMESTAMP]),
+    },
+    http: {
+      response: {
+        status_code: normalizeValue<number>(fields[HTTP_RESPONSE_STATUS_CODE]),
+      },
+    },
+    data_stream: {
+      namespace: normalizeValue<string>(fields[DATA_STREAM_NAMESPACE]),
+      type: normalizeValue<string>(fields[DATA_STEAM_TYPE]),
+      dataset: normalizeValue<string>(fields[DATA_STREAM_DATASET]),
+    },
+  };
+};
+
+export const spanLinksDetailsMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  return {
+    trace: {
+      id: normalizeValue<string>(fields[TRACE_ID]),
+    },
+    span: {
+      id: normalizeValue<string>(fields[SPAN_ID]),
+      name: normalizeValue<string>(fields[SPAN_NAME]),
+      type: normalizeValue<string>(fields[SPAN_TYPE]),
+      subtype: normalizeValue<string>(fields[SPAN_SUBTYPE]),
+      duration: {
+        us: normalizeValue<number>(fields[SPAN_DURATION]),
+      },
+    },
+    transaction: {
+      id: normalizeValue<string>(fields[TRANSACTION_ID]),
+      name: normalizeValue<string>(fields[TRANSACTION_NAME]),
+      duration: {
+        us: normalizeValue<string>(fields[TRANSACTION_DURATION]),
+      },
+    },
+    service: {
+      name: normalizeValue<string>(fields[SERVICE_NAME]),
+      environment: normalizeValue<string>(fields[SERVICE_ENVIRONMENT]),
+    },
+    processor: {
+      event: normalizeValue<string>(fields[PROCESSOR_EVENT]),
+    },
+    agent: {
+      name: normalizeValue<AgentName>(fields[AGENT_NAME]),
+    },
+  };
+};
+
+// todo: pending #192337
+export const linkedParentsOfSpanMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  return {
+    span: {
+      links: [
+        {
+          trace: {
+            id: normalizeValue<string>(fields[SPAN_LINKS_TRACE_ID]),
+          },
+          span: {
+            id: normalizeValue<string>(fields[SPAN_LINKS_SPAN_ID]),
+          },
+        },
+      ],
+    },
+  };
+};
+
+export const transactionMapping = (fields: Partial<Record<string, unknown[]>>) => {
```

Comment:
Can we make this returning `TransactionRaw` instead of `Record<string, unknown[]>`?

Reply:
Thank you for review @felixbarny! 

`Record<string, unknown[]>` is the parameter type, not return type. If the return type is missing, this means that I couldn't match it with any existing type interface, but typescript does automatically define an implicit type based on the return object format. So, we're left with correct typing at the end.

Today, I have a plan to check all mappings for a possible interface type that will match the returning type.